src/heatpumps/models/HeatPumpBase.py

```diff
@@ -948,3 +948,152 @@ def validate_dir(self):
+    #         else:
+    #             pass
+    #     else:
+    #         pass
+    #
+    #     if errors:
+    #         return errors
+    #     else:
+    #         return 'Die Simulation der Wärmepumpenauslegung war erfolgreich.'
+
+    def evap_state_condition_check(self, conn_valve_in, p_evap, wf):
+        """Checks the inlet and outlet state of evaporator valve. Inorder to avoid
+        the state after the expansion outside of the liquid vapor region.
+
+        arguments:
+            conn_valve_in   = The inlet connection name of the evaporator valve
+            p_evap          = The Evaporator pressure
+            wf              = working fluid"""
+
+        errors = set()
+
+        ### checks in and out of main evap valve
+        T_valve_in = self.conns[conn_valve_in].T.val
+        T_sat_evap = PSI(
+            'T', 'Q', 0, 'P', p_evap * 1e5,
+            wf) - 273.15
+
+        if T_valve_in > T_sat_evap:
+            pass
+        else:
+            errors.add(
+                f'Error: The temperature before the expansion {round(T_valve_in, 2)} °C '
+                f'should be greater than the saturation temperature {round(T_sat_evap, 2)}°C '
+                f'corresponding evaporator pressure.')
+
+        return errors
+
+    def mid_evap_state_condition_check(self, conn_mid_valve_in, p_mid, wf, **kwargs):
+        """Checks the inlet and outlet state of the mid valve. Inorder to avoid
+        the state after the expansion outside of the liquid vapor region.Also checks
+        the p_mid, which should be less than critical pressure of the working fluid chosen.
+
+        arguments:
+            conn_mid_valve_in   = The inlet connection name of the mid valve
+            p_mid          = The mid pressure
+            wf              = working fluid
+            kwargs :
+                econ = boolean
+                flash = boolean"""
+
+        errors = set()
+        p_crit = PSI('p_critical', wf) * 1e-5
+        T_mid_valve_in = self.conns[conn_mid_valve_in].T.val
+
+        if kwargs['econ']:
+            if self.econ_type == 'closed':
+                T_sat_p_mid = PSI(
+                    'T', 'Q', 0, 'P', p_mid * 1e5 / self.params['econ']['pr2'],
+                    wf) - 273.15
+            elif self.econ_type == 'open':
+                T_sat_p_mid = PSI(
+                    'T', 'Q', 0, 'P', p_mid * 1e5,
+                    wf) - 273.15
+            else:
+                raise ValueError(
+                    f"Parameter '{self.econ_type}' is not a valid econ_type. "
+                    + "Supported values are 'open' and 'closed'."
+                )
+        elif kwargs['flash']:
+            T_sat_p_mid = PSI(
+                'T', 'Q', 0, 'P', p_mid * 1e5,
+                wf) - 273.15
+        else:
+            pass
+
+        if T_mid_valve_in > T_sat_p_mid:
+            pass
+        else:
+            errors.add(f'Error: The temperature before mid expansion {round(T_mid_valve_in, 2)}°C '
```

Comment:
Is there a reason you are using `sets` and not `lists` or `tuples`? Also, why are you not concatinating strings with the `+` operator? Finally, you can round floats in f-strings with this notation: `f'... {T_mid_value_in:.2f}°C'`, with the number behind the dot setting the number of decimal places after the decimal point.

Reply:
I initially did it with **'list'**. At that time even if there is no error. I got error msg with with an empty list (**"[ ]"**). That means even if the simulation is Successful, gets error.